src/hugr.rs

```diff
@@ -74,29 +75,8 @@ impl Hugr {
     }
 
     /// Applies a rewrite to the graph.
-    pub fn apply_rewrite(mut self, rewrite: Rewrite) -> Result<(), RewriteError> {
-        // Get the open graph for the rewrites, and a HUGR with the additional components.
-        let (rewrite, mut replacement, parents) = rewrite.into_parts();
-
-        // TODO: Use `parents` to update the hierarchy, and keep the internal hierarchy from `replacement`.
-        let _ = parents;
-
-        let node_inserted = |old, new| {
-            std::mem::swap(&mut self.op_types[new], &mut replacement.op_types[old]);
-            // TODO: metadata (Fn parameter ?)
-        };
-        rewrite.apply_with_callbacks(
-            &mut self.graph,
-            |_| {},
-            |_| {},
-            node_inserted,
-            |_, _| {},
-            |_, _| {},
-        )?;
-
-        // TODO: Check types
-
-        Ok(())
+    pub fn apply<E>(&mut self, op: impl RewriteOp<E>) -> Result<(), E> {
```

Comment:
I'd keep the `apply_rewrite` name here, since it's the public API.
Otherwise the name won't mean much to the user.